src/utils/scroll-to/scroll-to.ts

```diff
@@ -0,0 +1,141 @@
+import { PluginObject } from 'vue';
+
+export enum ScrollToSpeed {
+    Slow = 'slow',
+    Slower = 'slower',
+    Regular = 'regular',
+    Fast = 'fast'
+}
+
+export enum ScrollToEasing {
+    Linear = 'linear',
+    EaseInQuad = 'easeInQuad',
+    EaseOutQuad = 'easeOutQuad',
+    EaseInCubic = 'easeInCubic',
+    EaseOutCubic = 'easeOutCubic'
+}
+
+// tslint:disable:typedef
+// linear
+const linear = t => t;
+// accelerating from zero velocity
+const easeInQuad = t => t * t;
+// decelerating to zero velocity
+const easeOutQuad = t => t * (2 - t);
+// accelerating from zero velocity
+const easeInCubic = t => t * t * t;
+// decelerating to zero velocity
+const easeOutCubic = t => --t * t * t + 1;
+// tslint:enable:typedef
+
+/**
+ *
+ */
+export class ScrollTo {
+
+    /**
+     * Scroll to the top of the page.
+     *
+     * @param offset
+     * @param speed
+     * @param easing
+     */
+    public goToTop(speed: ScrollToSpeed, easing: ScrollToEasing): Promise<any> {
+        const targetLocation: number = 0;
+        return this.internalScrollTo(targetLocation, speed, easing);
+    }
+
+    /**
+     * Scroll to the bottom of the page.
+     *
+     * @param offset
+     * @param speed
+     * @param easing
+     */
+    public goToBottom(speed: ScrollToSpeed, easing: ScrollToEasing): Promise<any> {
+        // const targetLocation: number = (window.innerHeight ||
+        //     (document.documentElement || document.body).clientHeight) + offset;
+
+        const targetLocation: number = document.body.offsetHeight ;
+
+        return this.internalScrollTo(targetLocation, speed, easing);
+    }
+
+    /**
+     * Scroll to a specified html element.
+     *
+     * @param offset
```

Comment:
peut-être documenter ce qu'il faut mettre comme valeur pour chaque param/pourquoi il est là.. : 
@param offset Offset from the top to define the relative initial position from where to calculate the scroll distance (when a sticky header is present for exemple)...;.

(ou quelque chose qui fait du sens..!)